core/primitives/src/sharding.rs

```diff
@@ -164,6 +167,26 @@ pub struct ShardChunk {
     pub header: ShardChunkHeader,
     pub transactions: Vec<SignedTransaction>,
     pub receipts: Vec<Receipt>,
+    /// Honeypot for catching lazy block producers
+    pub honeypot: Vec<Color>,
+}
+
+#[derive(
+    BorshSerialize, BorshDeserialize, Serialize, Deserialize, Debug, Clone, Copy, Eq, PartialEq,
+)]
+pub enum Color {
+    Red,
+    Blue,
+}
+
+impl Distribution<Color> for Standard {
+    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> Color {
+        if rng.gen_range(0, 2) == 0 {
```

Comment:
```suggestion
        if rng.gen() {
```

Reply:
https://docs.rs/rand/0.7.3/rand/trait.Rng.html#method.gen it seems that if I do `rng.gen()` it will generate a random integer.

Reply:
The return type of gen is parametrized. Just try changing and compiling it. I saw the code that does exactly I suggested.

Reply:
You might be even able to return `rng.gen()` without going through bool.